django/db/backends/postgresql/base.py

```diff
@@ -77,13 +83,13 @@ class DatabaseWrapper(BaseDatabaseWrapper):
         "BigAutoField": "bigint",
         "BinaryField": "bytea",
         "BooleanField": "boolean",
-        "CharField": "varchar(%(max_length)s)",
+        "CharField": _get_varchar_column,
         "DateField": "date",
         "DateTimeField": "timestamp with time zone",
         "DecimalField": "numeric(%(max_digits)s, %(decimal_places)s)",
         "DurationField": "interval",
-        "FileField": "varchar(%(max_length)s)",
-        "FilePathField": "varchar(%(max_length)s)",
+        "FileField": _get_varchar_column,
+        "FilePathField": _get_varchar_column,
```

Comment:
I'm not sure if unlimited file fields are necessary, most (all?) of filesystem have natural filename length limits. I'd revert this.
```suggestion
        "FileField": "varchar(%(max_length)s)",
        "FilePathField": "varchar(%(max_length)s)",
```

Reply:
Personally I think it's not the database's place to set these kind of restrictions, I would rather have the database not care and simply use unlimited varchars for everything and if you need to restrict the size do it higher up the stack with forms, serializers, etc. 

That said you can still set a database-level limit if you so desire, but then if the max length changes in the future you have to deal with migrations and potential table rewrites.

If you really think it should be kept as-is I'll revert.

Reply:
Personally, I think this ticket should be limited to the `CharField`. I've already presented my argument against unlimited `FileField`, `FilePathField`, and `SlugField`. Django doesn't need to have builtin support for all edge cases, folks can always add custom field types in their code.